Added support for "instanceTypesFilters" in ocean-aws-launchSpec, fixed "is_ena_supported"under filter in resource ocean-aws and moved credentials document under Accounts hierarchy.

sharadkesarwani opened the pull request:
Added support for "instanceTypesFilters" in  resource ocean-aws-launchSpec, fixed "is_ena_supported"under filter in resource ocean-aws and moved credentials document under Accounts hierarchy.

# Jira Ticket
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-15568

sharadkesarwani committed:
Added support for "instanceTypesFilters" in ocean-aws-launchSpec and fixed "is_ena_supported" in resource ocean-aws

Added support for "instanceTypesFilters" in ocean-aws-launchSpec, fixed "is_ena_supported"under filter in resource ocean-aws and moved credentials document under Accounts hierarchy.

sharadkesarwani committed:
Upgraded provider dependencies.

Upgraded provider dependencies.

chandra1-n commented on spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go:
```diff
@@ -1685,6 +1685,177 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
+					value = instanceTypesFilters
+				}
+			}
+			/*if cluster.Compute.InstanceTypes == nil {
```
remove the commented lines

sharadkesarwani replied:
Removed commented code

chandra1-n reviewed (changes requested):
please fix the review comments

sharadkesarwani committed:
Added documentation for "instancetypes_filters" under launchSpec for resource spotinst_ocean_aws_launch_spec and updated changelog.

Added documentation for "instancetypes_filters" under launchSpec for resource spotinst_ocean_aws_launch_spec and updated changelog.

sharadkesarwani committed:
removed commented code

removed commented code

chandra1-n commented on docs/resources/ocean_aws_launch_spec.md:
```diff
@@ -226,6 +247,31 @@ The following arguments are supported:
     * `http_put_response_hop_limit` - (Optional) An integer from 1 through 64. The desired HTTP PUT response hop limit for instance metadata requests. The larger the number, the further the instance metadata requests can travel.
+* `instancetypes_filters` - List of instance types filters. The instance types that match with all filters compose the Virtual Node Group's instanceTypes parameter.
+The architectures that come from the Virtual Node Group's images will be taken into account when using this parameter.
+Cannot be configured together with Virtual Node Group's instanceTypes and with the Cluster's whitelist/blacklist/filters.
```
continue this line from previous statement

sharadkesarwani replied:
Updated

chandra1-n commented on docs/resources/ocean_aws_launch_spec.md:
```diff
@@ -226,6 +247,31 @@ The following arguments are supported:
+* `instancetypes_filters` - List of instance types filters. The instance types that match with all filters compose the Virtual Node Group's instanceTypes parameter.
+The architectures that come from the Virtual Node Group's images will be taken into account when using this parameter.
+Cannot be configured together with Virtual Node Group's instanceTypes and with the Cluster's whitelist/blacklist/filters.
+    * `categories` - (Optional) Array of strings. The filtered instance types will belong to one of the categories types from this list. Valid values: `Accelerated_computing`, `Compute_optimized`,`General_purpose`,`Memory_optimized`, or `Storage_optimized`.
```
please remove data types from all parameters

remove "or" from valid values

sharadkesarwani replied:
removed "or"

chandra1-n commented on docs/resources/ocean_aws_launch_spec.md:
```diff
@@ -226,6 +247,31 @@ The following arguments are supported:
+The architectures that come from the Virtual Node Group's images will be taken into account when using this parameter.
+Cannot be configured together with Virtual Node Group's instanceTypes and with the Cluster's whitelist/blacklist/filters.
+    * `categories` - (Optional) Array of strings. The filtered instance types will belong to one of the categories types from this list. Valid values: `Accelerated_computing`, `Compute_optimized`,`General_purpose`,`Memory_optimized`, or `Storage_optimized`.
+    * `disk_types` - (Optional) Array of strings. The filtered instance types will have one of the disk type from this list. Valid values: `NVMe`, `EBS`, `SSD`, or `HDD`.
```
remove "or" from valid values

sharadkesarwani replied:
removed "or"

chandra1-n commented on docs/resources/ocean_aws_launch_spec.md:
```diff
@@ -226,6 +247,31 @@ The following arguments are supported:
+    * `exclude_families` - (Optional) Array of strings. Types belonging to a family from the ExcludeFamilies will not be available for scaling (asterisk wildcard is also supported). For example, C* will exclude instance types from these families: c5, c4, c4a, etc.
+    * `exclude_metal` - (Optional, Default: `false`)	
+Boolean. In case excludeMetal is set to true, metal types will not be available for scaling.
+    * `hypervisor` - (Optional) Array of strings. The filtered instance types will have a hypervisor type from this list. Valid values: `nitro`, or `xen`.
```
remove "or" from valid values

sharadkesarwani replied:
removed "or"

chandra1-n commented on docs/resources/ocean_aws_launch_spec.md:
```diff
@@ -226,6 +247,31 @@ The following arguments are supported:
+    * `disk_types` - (Optional) Array of strings. The filtered instance types will have one of the disk type from this list. Valid values: `NVMe`, `EBS`, `SSD`, or `HDD`.
+    * `exclude_families` - (Optional) Array of strings. Types belonging to a family from the ExcludeFamilies will not be available for scaling (asterisk wildcard is also supported). For example, C* will exclude instance types from these families: c5, c4, c4a, etc.
+    * `exclude_metal` - (Optional, Default: `false`)	
+Boolean. In case excludeMetal is set to true, metal types will not be available for scaling.
```
continue in the same line

sharadkesarwani replied:
In the same line now

chandra1-n commented on docs/resources/ocean_aws_launch_spec.md:
```diff
@@ -226,6 +247,31 @@ The following arguments are supported:
+    * `hypervisor` - (Optional) Array of strings. The filtered instance types will have a hypervisor type from this list. Valid values: `nitro`, or `xen`.
+    * `include_families` - (Optional) Array of strings. Types belonging to a family from the IncludeFamilies will be available for scaling (asterisk wildcard is also supported). For example, C* will include instance types from these families: c5, c4, c4a, etc.
+    * `is_ena_supported` - (Optional) 	
+Boolean. Ena is supported or not.
```
continue in the same line

sharadkesarwani replied:
In the same line now

chandra1-n commented on docs/resources/ocean_aws_launch_spec.md:
```diff
@@ -226,6 +247,31 @@ The following arguments are supported:
+    * `min_network_performance` - (Optional, >=0) Int. Minimum Bandwidth in Gib/s of network performance.
+    * `min_vcpu` - (Optional, >=0) Int. Minimum number of vcpus available.
+    * `root_device_types` - (Optional) Array of strings. The filtered instance types will have a root device types from this list.
+    Valid values: `ebs`, or `instance-store`.
```
continue in the same line

sharadkesarwani replied:
in the same line now

chandra1-n commented on docs/resources/ocean_aws_launch_spec.md:
```diff
@@ -226,6 +247,31 @@ The following arguments are supported:
+    * `min_vcpu` - (Optional, >=0) Int. Minimum number of vcpus available.
+    * `root_device_types` - (Optional) Array of strings. The filtered instance types will have a root device types from this list.
+    Valid values: `ebs`, or `instance-store`.
+    * `virtualization_types` - (Optional) Array of strings. The filtered instance types will support at least one of the virtualization types from this list. Valid values: `hvm`, or `paravirtual`.
```
remove "or" from valid values

sharadkesarwani replied:
removed "or"

chandra1-n reviewed (changes requested):
please fix the review comments

sharadkesarwani committed:
Removed data type for the fields and corrected formatting.

Removed data type for the fields and corrected formatting.

sharadkesarwani committed:
Updated document .

Removed "List of instance types filters" statement from "instancetypes_filters" description.
